baseapp/baseapp.go

```diff
@@ -57,15 +62,17 @@ var _ abci.Application = (*BaseApp)(nil)
 
 // Create and name new BaseApp
 // NOTE: The db is used to store the version number for now.
-func NewBaseApp(name string, cdc *wire.Codec, logger log.Logger, db dbm.DB) *BaseApp {
+func NewBaseApp(name string, cdc *wire.Codec, logger log.Logger, db dbm.DB, txGasLimit sdk.Gas) *BaseApp {
```

Comment:
Seems strange that txGasLimit would become so first class as to be an arg to NewBaseApp - is there a good reason for this ? 

Also, BaseApp should probably have a ConsensusParams that mirrors Tendermint's - MaxGasPerTx is part of that , gets initialized by InitChain, and can be updated by EndBlock. Then we shouldnt need to pass txGasLimit in everywhere.

In the meantime anyways I'd rather use setters and defaults than add function args

Reply:
Agreed on `ConsensusParams`, made an issue: https://github.com/cosmos/cosmos-sdk/issues/1007

Reply:
Removed `txGasLimit`, if Tendermint enforces that we may not need it at all actually.